src/Bicep.LangServer/Handlers/BicepHoverHandler.cs

```diff
@@ -36,6 +37,10 @@ public BicepHoverHandler(ISymbolResolver symbolResolver)
             {
                 return Task.FromResult<Hover?>(null);
             }
+            if (markdown.Length > LanguageConstants.MaxHoverMarkdownLength && markdown.EndsWith("```"))
```

Comment:
>EndsWith [](http://example.com/codeflow?start=87&length=8)

The dependency on the three trailing tickmarks could be easily missed and break the logic in the future. Also, the logic is currently eating the `\n` before the terminating tickmarks, which IIRC breaks the markdown codeblock formatting in VS code.

Can we change it so we have a `TruncateContent(string value, int limit)` method that truncates the `value` and adds the horizontal ellipsis and then call it in all the places in `GetMarkdown()`?

For example L62-63 below could look like this:
```csharp
case ParameterSymbol parameter:
  return CodeBlock(TruncateContent($"param {parameter.Name}: {parameter.Type}", `MaxHoverMarkdownLength`))

```

The above would also need a `CodeBlock(string content)` function to return something like this:
```csharp
return $"```bicep\n{content}\n"
```

I would also move the `MaxHoverMarkdownLength` from `LanguageConstants` to this class because it's only used here. It's specific to hovers, so it's not likely to be referenced outside of that. (I also wouldn't worry too much about the limit being exactly accurate. A couple of characters here don't really matter one way or another.)

Reply:
It's not about truncating content inside code block. it's about closing code block if we need to truncate content. for example in PropertySymbol code block is at the beginning, then it has series of lines with each property specified. If it will be too long - VSCode will add ellipsis by itself. We need to add it manually only if we need to close a code block, therefore I just combined those 2 functions into a single CodeBlock function with a limit.